### DISCUSSION TITLE
React-native error TypeError: undefined is not an object (evaluating 'globalThis.localStorage.getItem')

### DISCUSSION THREAD

**Discussion by 0verread** [supabase/supabase#8981] [General]
Trying to use supabase with a React-native app, get keep getting this error. I searched but couldn't find any good solution out there that is working for me.

>  ERROR  [TypeError: undefined is not an object (evaluating 'globalThis.localStorage.getItem')]

`import AsyncStorage from "@react-native-async-storage/async-storage";
import 'react-native-url-polyfill/auto';
import { createClient } from "@supabase/supabase-js";

import {REACT_NATIVE_SUPABASE_URL, REACT_NATIVE_SUPABASE_ANON_KEY} from '@env';

const supabaseUrl = REACT_NATIVE_SUPABASE_URL
const supabaseAnonKey = REACT_NATIVE_SUPABASE_ANON_KEY

const supabase = createClient(supabaseUrl, supabaseAnonKey, {
  localStorage: AsyncStorage
});

export default supabase;`

This is my supabase service code.

This might an old issue but I couldn't find any solution to this. Any help or ref will be appreciated.

**Comment by Hallidayo** [comment]
Hi @0verread,

Do you know what version of supabase you are using?  If you are using version 2 you could you at this:

https://github.com/supabase/examples/blob/main/supabase-js-v2/user-management/expo-user-management/lib/supabase.ts

**Reply by 0verread** [reply]
@Hallidayo , thanks for the response. I'm using supabase version 2- 

`"@supabase/supabase-js": "^2.0.0-rc.10",`

Also, I tried the above solution, but it didn't work. It is still throwing me the same error.

Also, I'm using react-native with JS, so is there something that's not supported in JS?

**Reply by 0verread** [reply]
This is solved.

The solution code :

`import AsyncStorage from "@react-native-async-storage/async-storage";
import { createClient } from "@supabase/supabase-js";
import { setupURLPolyfill } from 'react-native-url-polyfill';

setupURLPolyfill()
import {REACT_NATIVE_SUPABASE_URL, REACT_NATIVE_SUPABASE_ANON_KEY} from "@env";


const supabaseUrl = REACT_NATIVE_SUPABASE_URL
const supabaseAnonKey = REACT_NATIVE_SUPABASE_ANON_KEY

const supabase = createClient(supabaseUrl, supabaseAnonKey, {
  auth: {
    storage: AsyncStorage,
    persistSession: true,
    detectSessionInUrl: false,
    autoRefreshToken: true
  }
});

export default supabase;`
